app/views/dashboards/show.html.erb

```diff
@@ -0,0 +1,20 @@
+<h1><%= @environment.name %></h1>
+
+<section>
+  <table class="table">
+    <thead>
+    <tr>
+      <th></th>
+      <% @environment.deploy_groups.each do |deploy_group| %>
+        <th><%= deploy_group.name %></th>
+      <% end %>
+    </tr>
+    </thead>
+    <tbody>
+    <% @data.each do |project, deploy_group_versions| %>
+      <%= render partial: 'project', object: project,
```

Comment:
you might be able to get away with `render project, locals: { deploy_group_versions: deploy_group_versions }`